aten/src/ATen/native/vulkan/api/Context.cpp

```diff
@@ -41,7 +41,12 @@ VkDevice create_device(
       &device_extension_properties_count,
       device_extension_properties.data()));
 
+  #ifdef _MSC_VER
+  // MSVC does not support VLAs (variable length arrays).
+  std::vector<const char*> requested_device_extensions{
```

Comment:
OK this is not a C++14 problem and it's a bit more complicated than I thought. Turns out MSVC does not support VLAs (even without `constexpr`), so I have to use `std::vector` here for MSVC.